src/OpenSage.Game/Logic/Object/GameObject.cs

```diff
@@ -1476,7 +1476,7 @@ private void ExecuteRandomSlowDeathBehavior(DeathType deathType)
             .ToList();
 
         var sumProbabilityModifiers = slowDeathBehaviors.Sum(x => x.ProbabilityModifier);
-        var random = _gameEngine.Random.Next(sumProbabilityModifiers);
+        var random = _gameEngine.GameLogic.Random.Next(0, sumProbabilityModifiers);
```

Comment:
does this need to be `sumProbabilityModifiers - 1`? Not sure of the context here, figured I'd check to be safe